x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/lucene/ValueSourceReaderTypeConversionTests.java

```diff
@@ -0,0 +1,2021 @@
+        try {
+            for (String indexKey : INDICES.keySet()) {
+                initIndex(indexKey, size, commitEvery);
+                shardContexts.add(new LuceneSourceOperatorTests.MockShardContext(reader(indexKey), INDICES.get(indexKey).shardIdx));
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        var luceneFactory = new LuceneSourceOperator.Factory(
+            shardContexts,
+            ctx -> new MatchAllDocsQuery(),
+            DataPartitioning.SHARD,
+            1,// randomIntBetween(1, 10),
+            pageSize,
+            LuceneOperator.NO_LIMIT
+        );
+        return luceneFactory.get(context);
+    }
+
+    private void initMapping(String indexKey) throws IOException {
+        TestIndexMappingConfig indexMappingConfig = INDICES.get(indexKey);
+        mapperServices.put(indexKey, new MapperServiceTestCase() {
+        }.createMapperService(MapperServiceTestCase.mapping(b -> {
+            fieldExamples(b, "key", "integer"); // unique key per-index to use for looking up test values to compare to
+            fieldExamples(b, "indexKey", "keyword");  // index name (can be used to choose index-specific test values)
+            fieldExamples(b, "int", "integer");
+            fieldExamples(b, "short", "short");
+            fieldExamples(b, "byte", "byte");
+            fieldExamples(b, "long", "long");
+            fieldExamples(b, "double", "double");
+            fieldExamples(b, "kwd", "keyword");
+            b.startObject("stored_kwd").field("type", "keyword").field("store", true).endObject();
+            b.startObject("mv_stored_kwd").field("type", "keyword").field("store", true).endObject();
+
+            simpleField(b, "missing_text", "text");
+
+            for (Map.Entry<String, TestFieldType<?>> entry : indexMappingConfig.fieldTypes.entrySet()) {
+                String fieldName = entry.getKey();
+                TestFieldType<?> fieldType = entry.getValue();
+                simpleField(b, fieldName, fieldType.dataType.typeName());
+            }
+        })));
+    }
+
+    private void initIndex(String indexKey, int size, int commitEvery) throws IOException {
+        initMapping(indexKey);
+        readers.put(indexKey, initIndex(indexKey, directory(indexKey), size, commitEvery));
+    }
+
+    private IndexReader initIndex(String indexKey, Directory directory, int size, int commitEvery) throws IOException {
+        keyToTags.computeIfAbsent(indexKey, k -> new HashMap<>()).clear();
+        TestIndexMappingConfig indexMappingConfig = INDICES.get(indexKey);
+        try (
+            IndexWriter writer = new IndexWriter(
+                directory,
+                newIndexWriterConfig().setMergePolicy(NoMergePolicy.INSTANCE).setMaxBufferedDocs(IndexWriterConfig.DISABLE_AUTO_FLUSH)
+            )
+        ) {
+            for (int d = 0; d < size; d++) {
```

Comment:
This code is highly specific to this test - it would be useful to have a description of what the docs look like since there's no templating or external file to look at.